plugin/pkg/admission/limitranger/admission.go

```diff
@@ -210,76 +210,107 @@ func NewLimitRanger(actions LimitRangerActions) (*LimitRanger, error) {
 	}, nil
 }
 
-// defaultContainerResourceRequirements returns the default requirements for a container
+type defaultResources struct {
+	pod api.ResourceRequirements
+	container api.ResourceRequirements
+}
+
+// defaultResourceRequirements returns the default requirements for container,
+// if PodLevelResources is enabled, returns the default requirements for pod
 // the requirement.Limits are taken from the LimitRange defaults (if specified)
 // the requirement.Requests are taken from the LimitRange default request (if specified)
-func defaultContainerResourceRequirements(limitRange *corev1.LimitRange) api.ResourceRequirements {
-	requirements := api.ResourceRequirements{}
-	requirements.Requests = api.ResourceList{}
-	requirements.Limits = api.ResourceList{}
+func defaultResourceRequirements(limitRange *corev1.LimitRange, podLevelResourcesEnabled bool) *defaultResources {
+	defaultResources := defaultResources{}
 
 	for i := range limitRange.Spec.Limits {
 		limit := limitRange.Spec.Limits[i]
-		if limit.Type == corev1.LimitTypeContainer {
-			for k, v := range limit.DefaultRequest {
-				requirements.Requests[api.ResourceName(k)] = v.DeepCopy()
-			}
-			for k, v := range limit.Default {
-				requirements.Limits[api.ResourceName(k)] = v.DeepCopy()
-			}
+
+		if limit.Type != corev1.LimitTypeContainer && limit.Type != corev1.LimitTypePod || !podLevelResourcesEnabled && limit.Type == corev1.LimitTypePod {
```

Comment:
Can we possible make this if condition a bit more readable and also add a comment

Reply:
Done, added comments and separated the condition into two if statements for better readability.